File: app/src/main/java/fr/abouveron/projectamio/MainService.java
package fr.abouveron.projectamio;

import android.app.AlarmManager;
import android.app.PendingIntent;
import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.os.Handler;
import android.os.IBinder;
import android.os.Looper;
import android.os.SystemClock;
import android.util.Log;
import android.widget.Toast;

public class MainService extends Service {
    private Handler handler;
    private Runnable runnable;

    @Override
    public void onCreate() {
        super.onCreate();

        Toast.makeText(
                getApplicationContext(),
                "Service started",
                Toast.LENGTH_SHORT
        ).show();

        handler = new Handler(Looper.getMainLooper());
        runnable = new Runnable() {
            @Override
            public void run() {
                new WebService(getApplicationContext()).execute();
                handler.postDelayed(this, 5000);
            }
        };

        handler.post(runnable);
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        Log.d("MainService", "onStartCommand");
        return START_STICKY;
    }

    @Override
    public void onTaskRemoved(Intent rootIntent) {
        Intent restartServiceIntent = new Intent(getApplicationContext(), getClass());
        restartServiceIntent.setPackage(getPackageName());
        PendingIntent restartServicePendingIntent = PendingIntent.getService(
                getApplicationContext(), 1, restartServiceIntent, PendingIntent.FLAG_ONE_SHOT | PendingIntent.FLAG_IMMUTABLE);

        AlarmManager alarmService = (AlarmManager) getSystemService(Context.ALARM_SERVICE);
        if (alarmService != null) {
            alarmService.set(AlarmManager.ELAPSED_REALTIME, SystemClock.elapsedRealtime() + 1000, restartServicePendingIntent);
        }

        super.onTaskRemoved(rootIntent);
    }


    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        handler.removeCallbacks(runnable);
        Log.d("MainService", "Service destroyed");
        Toast.makeText(
                getApplicationContext(),
                "Service stopped",
                Toast.LENGTH_SHORT
        ).show();
    }
}

File: app/src/main/java/fr/abouveron/projectamio/MainService.kt
package fr.abouveron.projectamio

import android.app.Service
import android.content.Intent
import android.os.IBinder
import android.util.Log
import android.widget.Toast
import kotlinx.coroutines.*

class MainService : Service() {
    private var temps = 0
    private var job: Job? = null

    override fun onCreate() {
        super.onCreate()
        temps = 0

        job = CoroutineScope(Dispatchers.Main).launch {
            while (isActive) {
                withContext(Dispatchers.Main) {
                    Toast.makeText(
                        applicationContext,
                        temps.toString(),
                        Toast.LENGTH_SHORT
                    ).show()
                    Log.d("MainService", temps.toString())
                    delay(5000) // 5 second delay
                    temps += 5
                }
            }
        }
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        Log.d("MainService", "onStartCommand")
        return START_STICKY
    }

    override fun onBind(p0: Intent?): IBinder? {
        return null
    }

    override fun onDestroy() {
        super.onDestroy()
        job?.cancel()
        Log.d("MainService", "Service destroyed")
    }
}
